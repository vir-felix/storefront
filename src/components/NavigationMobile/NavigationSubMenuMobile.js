import React, { Component } from "react";
import PropTypes from "prop-types";
import Divider from "@material-ui/core/Divider";
import Toolbar from "@material-ui/core/Toolbar";
import IconButton from "@material-ui/core/IconButton";
import Typography from "@material-ui/core/Typography";
import MenuList from "@material-ui/core/MenuList";
import ChevronLeftIcon from "mdi-material-ui/ChevronLeft";
import { withStyles } from "@material-ui/core/styles";
import NavigationItemMobile from "./NavigationItemMobile";

const styles = () => ({
  root: {
    display: "flex",
    flexDirection: "column"
  },
  header: {
    flex: "0 0 auto"
  },
  menu: {
    flex: "1 1 auto",
    overflowY: "auto"
  }
});

@withStyles(styles, { name: "SkNavigationItemMobile" })
class NavigationSubMenuMobile extends Component {
  static propTypes = {
    classes: PropTypes.object,
    navItem: PropTypes.object,
    onBackButtonClick: PropTypes.func,
    routingStore: PropTypes.object
  };

  static defaultProps = {
    classes: {},
    navItem: {},
    onBackButtonClick() {},
    routingStore: {}
  };

  state = { isSubNavOpen: false };

  get hasSubNavItems() {
    const { navItem: { subTags } } = this.props;
    return Array.isArray(subTags.edges) && subTags.edges.length > 0;
  }

  render() {
    const { classes, navItem } = this.props;

    if (!navItem) {
      return null;
    }

    return (
      <div className={classes.root}>
        <div className={classes.header}>
          <Toolbar disableGutters>
            <IconButton onClick={this.props.onBackButtonClick}>
              <ChevronLeftIcon />
            </IconButton>
            <Typography variant="subheading">{navItem.name}</Typography>
          </Toolbar>
          <Divider />
        </div>
        <div className={classes.menu}>
          {this.hasSubNavItems &&
            <MenuList component="div" disablePadding>
              {navItem.subTags.edges.map(({ node: navItemGroup }, index) => (
                <NavigationItemMobile key={index} navItem={navItemGroup} />
              ))}
            </MenuList>
          }
        </div>
      </div>
    );
  }
}

export default NavigationSubMenuMobile;
