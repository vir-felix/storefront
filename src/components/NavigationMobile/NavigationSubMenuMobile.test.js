import React from "react";
import renderer from "react-test-renderer";
import { Provider } from "mobx-react";
import NavigationSubMenuMobile from "./NavigationSubMenuMobile";

const mockTagWithSubTags = {
  _id: "cmVhY3Rpb24vdGFnOjVXZE5LYXprejZ5TWdrcGY5",
  position: null,
  slug: "tag-slug",
  name: "Tag Name",
  subTags: {
    edges: [
      {
        node: {
          _id: "cmVhY3Rpb24vdGFnOlRFdjdSV1pDTHl1d1dmOWs2",
          position: null,
          slug: "sub-tag-slug",
          name: "Sub Tag Name",
          subTags: {
            edges: []
          }
        }
      },
      {
        node: {
          _id: "cmVhY3Rpb24vdGFnOmJ2MmVEbzV3V3lvNUdNMlN3",
          position: null,
          slug: "sub-tag-slug",
          name: "Sub Tag name",
          subTags: {
            edges: []
          }
        }
      }
    ]
  }
};

const uiStore = {
  toggleMenuDrawerOpen() {}
};

const routingStore = {
  tag: {
    _id: "v2hE7fzL6cJniwgSm",
    name: "Tag A",
    slug: "tag-a",
    shopId: "J8Bhq3uTtdgwZx3rz",
    isTopLevel: true,
    updatedAt: "2018-06-11T18:01:09.568Z",
    createdAt: "2018-06-11T18:00:48.414Z",
    isDeleted: false,
    isVisible: true,
    groups: [],
    relatedTagIds: [
      "s3nybDz4QPkmapETc",
      "JB4FRBiWduNsxhhhn",
      "gRM3ADcY77eNJ7Brm"
    ]
  }
};

test("basic snapshot", () => {
  const component = renderer.create((
    <Provider primaryShopId={"J8Bhq3uTtdgwZx3rz"} routingStore={routingStore} uiStore={uiStore}>
      <NavigationSubMenuMobile navItem={mockTagWithSubTags} />
    </Provider>
  ));
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
