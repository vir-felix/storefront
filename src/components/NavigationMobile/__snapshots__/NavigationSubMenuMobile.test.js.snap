// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`basic snapshot 1`] = `
<div
  className="SkNavigationItemMobile-root-1"
>
  <div
    className="SkNavigationItemMobile-header-2"
  >
    <div
      className="MuiToolbar-root-4 MuiToolbar-regular-6"
    >
      <button
        className="MuiButtonBase-root-14 MuiIconButton-root-8"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="0"
        type="button"
      >
        <span
          className="MuiIconButton-label-13"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-17"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M15.41,16.58L10.83,12L15.41,7.41L14,6L8,12L14,18L15.41,16.58Z"
            />
          </svg>
        </span>
        <span
          className="MuiTouchRipple-root-26"
        />
      </button>
      <h3
        className="MuiTypography-root-33 MuiTypography-subheading-40"
      >
        Tag Name
      </h3>
    </div>
    <hr
      className="MuiDivider-root-59"
    />
  </div>
  <div
    className="SkNavigationItemMobile-menu-3"
  >
    <div
      className="MuiList-root-63"
      onBlur={[Function]}
      onKeyDown={[Function]}
      role="menu"
    >
      <div
        className="MuiButtonBase-root-14 MuiListItem-root-71 SkNavigationItemMobile-listItemRoot-68 MuiListItem-default-74 MuiListItem-gutters-78 MuiListItem-button-79"
        color="inherit"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex="0"
      >
        <div
          className="MuiListItemText-root-82"
        >
          <span
            className="MuiTypography-root-33 MuiTypography-subheading-40 MuiListItemText-primary-85"
          >
            Sub Tag Name
          </span>
        </div>
        <span
          className="MuiTouchRipple-root-26"
        />
      </div>
      <hr
        className="MuiDivider-root-59"
      />
      <div
        className="MuiButtonBase-root-14 MuiListItem-root-71 SkNavigationItemMobile-listItemRoot-68 MuiListItem-default-74 MuiListItem-gutters-78 MuiListItem-button-79"
        color="inherit"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex="0"
      >
        <div
          className="MuiListItemText-root-82"
        >
          <span
            className="MuiTypography-root-33 MuiTypography-subheading-40 MuiListItemText-primary-85"
          >
            Sub Tag name
          </span>
        </div>
        <span
          className="MuiTouchRipple-root-26"
        />
      </div>
      <hr
        className="MuiDivider-root-59"
      />
    </div>
  </div>
</div>
`;
